### ISSUE TITLE
clearMediaCollection and getFirstMediaUrl id discrepancy

### ISSUE THREAD

**Issue by kickthemooon** [spatie/laravel-medialibrary#398]
so this is my controller method:
```
$file = $request->file('file');
        $settings = Settings::first();
        $settings->clearMediaCollection('main_header_logo');
        $settings->addMedia($file)->toCollection('main_header_logo');
        $image = $settings->getFirstMediaUrl('main_header_logo', 'main_header_logo');
        return response([
            'status' => 'success',
            'imageurl'   => $image
        ]);
```
So the process is:
1. Clear media collection
2. Add media to collection
3. Get first (and assuming only) media url

The problem:
getFirstMediaUrl or getFirstMedia gives me the id of the media that was there before i cleared the collection
so for example, it gives me 99 when actually the new added media is 100.

**Comment by freekmurze** [comment]
When I have a little time on my hands, I'll add a test for this case to our testsuite (and fix it if that proves that there is indeed a problem)

**Comment by kickthemooon** [comment]
not sure if it has something to do with event being fired:

```
public function clearMediaCollection(string $collectionName = 'default')
    {
        $this->getMedia($collectionName)->map(function (Media $media) {
            app(Filesystem::class)->removeFiles($media);
            $media->delete();
        });

        event(new CollectionHasBeenCleared($this, $collectionName));

        return $this;
    }
```

at the moment i am solving it like this:

```
$item = $object->addMedia($file)->toCollection($collectionName);
$media = Media::find($item['attributes']['id']);
```
so this gives me the exact media object so everything works fine

**Comment by kickthemooon** [comment]
@freekmurze the HasMediaTrait has this updateMedia() method. Am not sure but may this could help with this issue. Can you share an example of how to use this method, updateMedia()?

Otherwise I tried to return false in the MediaRepository mediaIsPreloaded() and it works, but no sure if thats good...

**Comment by emilioborraz** [comment]
I'm experiencing the same issue:

1. $emailNotification->clearMediaCollection();
2. $emailNotification->addMediaFromRequest('attachment')->usingName('email_notification_'.$emailNotification->id)->toMediaLibrary();
 3. return $emailNotification->getMedia()->first()

Returns the last deleted media.

**Comment by emilioborraz** [comment]
My solution to this so far is to unset the Media relationship, please see step 3.

1. $emailNotification->clearMediaCollection();
2. $emailNotification->addMediaFromRequest('attachment')->usingName('email_notification_'.$emailNotification->id)->toMediaLibrary();
3. $emailNotification->media = null;
4. return $emailNotification->getMedia()->first();

**Comment by freekmurze** [comment]
This is fixed in the newly tagged version `4.10.2`
